### PR TITLE
Terraform 1.2.0 upgrade

### PR DESCRIPTION
**Pull request by LeePorte**
What
----

Upgraded terraform to 1.2.0 and the modules to  the latest version so that we stay current

How to review
-------------

Code review and review the output of
https://deployer.dev03.dev.cloudpipeline.digital/teams/main/pipelines/create-cloudfoundry/jobs/cf-terraform/builds/58
https://deployer.dev03.dev.cloudpipeline.digital/teams/main/pipelines/create-cloudfoundry/jobs/psn-terraform/builds/56
https://deployer.dev03.dev.cloudpipeline.digital/teams/main/pipelines/create-cloudfoundry/jobs/vpc-peering-terraform/builds/57
https://deployer.dev03.dev.cloudpipeline.digital/teams/main/pipelines/create-cloudfoundry/jobs/az-healthcheck-terraform/builds/56

---

🚨⚠️ Please do not merge this pull request via the GitHub UI ⚠️🚨

### CODE REVIEW

File: concourse/pipelines/create-cloudfoundry.yml [hunk 1]
```diff
@@ -1286,6 +1286,10 @@ jobs:
                 . terraform-variables/cf-secrets.tfvars.sh
 
                 cp cf-tfstate/cf.tfstate updated-tfstate/cf.tfstate
+                terraform import aws_s3_bucket_lifecycle_configuration.droplets-s3 ((deploy_env))-cf-droplets
```
Comment (AP-Hunt) [inline concourse/pipelines/create-cloudfoundry.yml]: Do we need to undo this change after the state file has been changed?

File: concourse/tasks/aiven-broker-acceptance-tests-run.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ image_resource:
   type: docker-image
   source:
     repository: ghcr.io/alphagov/paas/cf-acceptance-tests
-    tag: 168a6c8b43fdec99607fd8bc01c5986a46a72dc0
+    tag: 7191ee87b98642e5d03a80f24585fddc02fc8d88
```
Comment (AP-Hunt) [inline concourse/tasks/aiven-broker-acceptance-tests-run.yml:7]: Could we merge the changes in this commit in to the first commit that changes the hashes?

File: concourse/pipelines/create-cloudfoundry.yml [hunk 3]
```diff
@@ -1286,10 +1286,6 @@ jobs:
                 . terraform-variables/cf-secrets.tfvars.sh
 
                 cp cf-tfstate/cf.tfstate updated-tfstate/cf.tfstate
-                terraform import aws_s3_bucket_lifecycle_configuration.droplets-s3 ((deploy_env))-cf-droplets
```
Comment (AP-Hunt) [inline concourse/pipelines/create-cloudfoundry.yml]: Why did we add and remove this in a single PR?

**Review by AP-Hunt — CHANGES_REQUESTED** [review]

File: concourse/pipelines/create-cloudfoundry.yml (hunk 1 above)
Comment (LeePorte) [reply concourse/pipelines/create-cloudfoundry.yml]: Nope

File: concourse/pipelines/create-cloudfoundry.yml (hunk 3 above)
Comment (LeePorte) [reply concourse/pipelines/create-cloudfoundry.yml]: It wasn't actually needed but was referenced in the upgrade documents

**Review by AP-Hunt — APPROVED** [review]
